libsplinter/src/rest_api/actix_impl.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Actix implemenation of the splinter REST API.
+
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::fmt;
+
+use actix_http::Payload;
+use actix_web::{web::Query, FromRequest, HttpRequest, ResponseError};
+
+use super::{Request, RequestBuilder};
+
+impl FromRequest for Request {
```

Comment:
_Obligatory rant about how trait implementations for a struct should be in the same file as the struct definition_

I would consider making Request a trait then having an Actix specific implementation that's defined in actix_impl.